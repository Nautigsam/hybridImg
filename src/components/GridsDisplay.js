import React from 'react';

import { Notification, Button } from 'react-bulma-components';
import { Link } from 'react-router-dom';

import StoreProvider from '../store/StoreProvider';
import ElementsDisplay from './ElementsDisplay';
import GridPreview from './GridPreview';

export const GridsDisplay = ({ grids, filter, byLine }) => {
  filter = filter || {};

  const filterGrids = grids.filter(g => {
    return (
      (!filter.user || filter.user === g.user) &&
      (!filter.tags ||
        filter.tags.every(
          t => g.lineThemes.includes(t) || g.lineThemes.includes(t)
        )) &&
      (!filter.name ||
        g.name.toLowerCase().match(`/.*${filter.name.toLowerCase()}.*/`))
    );
  });

  return (
    <>
      <Notification color="primary" className="is-light">
        <Link to="/grids/new">
          <Button color="primary">Nouvelle grille</Button>
        </Link>
      </Notification>
      <ElementsDisplay
        ChildComponent={GridPreview}
        getChildProps={grid => ({ id: grid.id })}
        elements={filterGrids}
        byLine={byLine}
      />
    </>
  );
};

function extraProps(store) {
  return {
    grids: Object.values(store.grids),
  };
}

export default StoreProvider(extraProps)(GridsDisplay);
