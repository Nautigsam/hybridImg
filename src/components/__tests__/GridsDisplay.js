import React from 'react';
import { shallow } from 'enzyme';

import { GridsDisplay } from '../GridsDisplay';

describe('GridsDisplay', () => {
  it('renders correctly', () => {
    const props = {
      grids: [{ id: 'G1' }, { id: 'G2' }],
      users: [{ id: 'u1', name: 'Mag' }, { id: 'u2', name: 'Matt' }],
    };
    const wrapper = shallow(<GridsDisplay {...props} />);
    expect(wrapper).toMatchSnapshot();
  });
});
