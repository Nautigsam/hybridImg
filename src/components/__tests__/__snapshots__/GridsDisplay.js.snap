// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GridsDisplay renders correctly 1`] = `
<Fragment>
  <Notification
    className="is-light"
    color="primary"
    renderAs="div"
  >
    <Link
      to="/grids/new"
    >
      <Button
        color="primary"
        disabled={false}
        fullwidth={false}
        inverted={false}
        isSelected={false}
        isStatic={false}
        loading={false}
        onClick={[Function]}
        outlined={false}
        remove={false}
        renderAs="button"
        reset={false}
        rounded={false}
        submit={false}
        text={false}
      >
        Nouvelle grille
      </Button>
    </Link>
  </Notification>
  <ElementsDisplay
    ChildComponent={
      Object {
        "$$typeof": Symbol(react.memo),
        "compare": null,
        "type": [Function],
      }
    }
    elements={
      Array [
        Object {
          "id": "G1",
        },
        Object {
          "id": "G2",
        },
      ]
    }
    getChildProps={[Function]}
  />
</Fragment>
`;
